Adapter/Adapter.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * @author Rafał Muszyński <rafal.muszynski@sourcefabric.org>
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt
+ */
+namespace Newscoop\PaywallBundle\Adapter;
+
+use Newscoop\PaywallBundle\Services\PaywallService;
+use Newscoop\PaywallBundle\Entity\OrderInterface;
+use Omnipay\Omnipay;
+
+/**
+ * Omnipay adapter.
+ */
+class Adapter
+{
+    protected $subscriptionService;
+
+    protected $request;
+
+    protected $gateway;
+
+    protected $router;
+
+    public function __construct(PaywallService $subscriptionService, $router, array $config = array())
+    {
+        $this->subscriptionService = $subscriptionService;
+        $this->config = $config;
+        $this->router = $router;
+        $this->initializeGateway();
```

Comment:
fetching any data from DB in constructor is bad idea, as on every request (even it's not needed) it will be called.